### PR TITLE
Landlord Listing Form + Unit Details Updates

### LINKED ISSUE #98: Landlord Form + Unit Details Updates
# Overview

During our client meeting, the client requested a few changes with the Landlord Form and the Unit Details page.

## Frontend

1. Update the Unit details page so that only HLs can update the HL column for a referral:

    In the below table, only show the HL dropdown if the current user is an HL. Otherwise, if they are an RS, just show the name of the HL as plain text.

    image

2. In the Landlord Listing Form, landlords will also need to tell us what utilities they pay for. Under the "Select All Appliances that Apply" section, add a required section called "Utilities Included" with the following checkbox options: Electricity, Water, Gas, Trash

3. Update the Unit details page to show a new "Utilities:" section under the "Appliances:" section and populate it with the data collected from the form.

## Backend

1. Update the backend to support the new Utilities section on the Landlord Listing Form

    - Update the `Unit` model to include the new field
    - Update the unit creation route validators to include the new field
    - .... and other necessary changes 

## Acceptance Criteria
- [x] Only HLs should be able to update assigned HL for a referral
- [x] Submitting the listing form with utilities saves to the database correctly
- [x] A new Utilities section is shown on the unit details page

### PR DESCRIPTION
**Pull request by YashRavipati1**
## Tracking Info

Resolves #98

## Changes

1. Updated Unit Details page such that only HLs can update the HL column for a referral
2. Added utilities component to landlord listing form and unit details

## Testing

Looked at a unit details page from both a HL account and a non-HL account to verify the referral table.
Went through entire workflow of landlord listing form to create unit - validated that utilities were successfully added and showing up on unit details.

## Confirmation of Change

Screenshot 2024-05-29 at 2 10 51 AM

HL Account:
Screenshot 2024-05-29 at 2 11 08 AM

Non-HL Account:
Screenshot 2024-05-29 at 2 11 51 AM

### CODE REVIEW

File: backend/src/validators/units.ts [hunk 1]
```diff
@@ -90,6 +90,7 @@ const createUnitSchema = [
   body("numBeds").exists().withMessage("is required").isNumeric().withMessage("must be a number"),
   body("numBaths").exists().withMessage("is required").isNumeric().withMessage("must be a number"),
   body("appliances").exists().withMessage("is required").isArray().withMessage("must be an array"),
+  body("utilities").exists().withMessage("is required").isArray().withMessage("must be an array"),
```
Comment (Azhou2023) [inline backend/src/validators/units.ts:93]: I'm not sure if this is necessary, but if this section is required but the landlord doesn't provide any of the the utilities can they still submit the form? Should there be an N/A section also or are they just not allowed to submit the unit if they don't provide at least one of those utilities?

Comment (Azhou2023) [reply backend/src/validators/units.ts:93]: Oh ok I see that you can actually submit without filling out the utilities section, but then should it be labeled with the required star?

**Review by Azhou2023 — CHANGES_REQUESTED** [review]
Looks good to me! Just one comment about maybe a potential N/A field for the utilities section?

**Review by petabite — APPROVED** [review]
LGTM 🔥 🔥

File: backend/src/validators/units.ts (hunk 1 above)
Comment (petabite) [reply backend/src/validators/units.ts:93]: Yeah that's for sure an inconsistency. This issue also applies to other fields on the form, so I think we should address this in another task. Thanks for bringing it up though!

**Review by Azhou2023 — APPROVED** [review]

**Pull request merged**